From 5fd115fb9e2210867854e0cf4d5bc77e26a9eaf1 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Thu, 15 Feb 2018 12:10:00 +0000
Subject: [PATCH 21/80] vc_sm: Correct kernel API call to return uint32_t for
 VC handle

The VC handle on an object is a uint32_t.
vc_sm_int_handle is meant to return the VC handle, but is
returning an int. Correct this to avoid issues.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/char/broadcom/vc_sm/vc_sm_knl.h | 2 +-
 drivers/char/broadcom/vc_sm/vmcs_sm.c   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/char/broadcom/vc_sm/vc_sm_knl.h b/drivers/char/broadcom/vc_sm/vc_sm_knl.h
index f7f74750d835..0eb71cbcd233 100644
--- a/drivers/char/broadcom/vc_sm/vc_sm_knl.h
+++ b/drivers/char/broadcom/vc_sm/vc_sm_knl.h
@@ -41,7 +41,7 @@ int vc_sm_lock(int handle, enum vc_sm_lock_cache_mode mode,
 int vc_sm_unlock(int handle, int flush, int no_vc_unlock);
 
 /* Get an internal resource handle mapped from the external one. */
-int vc_sm_int_handle(int handle);
+uint32_t vc_sm_int_handle(int handle);
 
 /* Map a shared memory region for use by kernel. */
 int vc_sm_map(int handle, unsigned int sm_addr,
diff --git a/drivers/char/broadcom/vc_sm/vmcs_sm.c b/drivers/char/broadcom/vc_sm/vmcs_sm.c
index 1bc37ee88225..11f23f91ac7c 100644
--- a/drivers/char/broadcom/vc_sm/vmcs_sm.c
+++ b/drivers/char/broadcom/vc_sm/vmcs_sm.c
@@ -3368,7 +3368,7 @@ int vc_sm_alloc(struct vc_sm_alloc_t *alloc, int *handle)
 EXPORT_SYMBOL_GPL(vc_sm_alloc);
 
 /* Get an internal resource handle mapped from the external one. */
-int vc_sm_int_handle(int handle)
+uint32_t vc_sm_int_handle(int handle)
 {
 	struct sm_resource_t *resource;
 	int ret = 0;
-- 
2.17.0

